common/rya.api/src/main/java/org/apache/rya/api/functions/OWLTime.java

```diff
@@ -0,0 +1,147 @@
+     */
+    public static final URI MINUTES_URI;
+    /**
+     * Hours class of type DurationDescription in OWL-Time ontology
+     */
+    public static final URI HOURS_URI;
+    /**
+     * Days class of type DurationDescription in OWL-Time ontology
+     */
+    public static final URI DAYS_URI;
+    /**
+     * Weeks class of type DurationDescription in OWL-Time ontology
+     */
+    public static final URI WEEKS_URI;
+    /**
+     * Months class of type DurationDescription in OWL-Time ontology
+     */
+    public static final URI MONTHS_URI;
+    /**
+     * Years class of type DurationDescription in OWL-Time ontology
+     */
+    public static final URI YEARS_URI;
+
+    public static boolean isValidDurationType(URI duration) {
+        return (duration.equals(SECONDS_URI) || duration.equals(MINUTES_URI) || duration.equals(HOURS_URI) || duration.equals(DAYS_URI)
+                || duration.equals(WEEKS_URI) || duration.equals(MONTHS_URI) || duration.equals(YEARS_URI));
+    }
+
+    static {
+        ValueFactory factory = ValueFactoryImpl.getInstance();
+        SECONDS_URI = factory.createURI(NAMESPACE, "seconds");
+        MINUTES_URI = factory.createURI(NAMESPACE, "minutes");
+        HOURS_URI = factory.createURI(NAMESPACE, "hours");
+        DAYS_URI = factory.createURI(NAMESPACE, "days");
+        WEEKS_URI = factory.createURI(NAMESPACE, "weeks");
+        MONTHS_URI = factory.createURI(NAMESPACE, "months");
+        YEARS_URI = factory.createURI(NAMESPACE, "years");
+    }
+
+    public static enum Duration {
+        SECONDS(1000, SECONDS_URI), 
+        MINUTES(60000, MINUTES_URI), 
+        HOURS(3600000, HOURS_URI), 
+        DAYS(86400000, DAYS_URI), 
+        WEEKS(604800000, WEEKS_URI), 
+        MONTHS(2592000000L, MONTHS_URI), 
+        YEARS(31536000000L, YEARS_URI);
+
+        private final URI uri;
+        private final long millisConversion;
+
+        Duration(long millisConversion, URI uri) {
+            this.uri = uri;
+            this.millisConversion = millisConversion;
+        }
+
+        public long millisConversion() {
+            return millisConversion;
+        }
+
+        public URI uri() {
+            return uri;
+        }
+
+        /**
+         * Returns the duration in milliseconds
+         * 
+         * @param duration - amount of time in the units indicated by the provided {@link OWLTime} URI
+         * @param uri - OWLTime URI indicating the time unit of duration
+         * @return - the amount of time in milliseconds 
+         */
+        public static long getMillis(int duration, URI uri) {
+            Duration durEnum = getDurationFromURI(uri);
+            return duration * durEnum.millisConversion();
+        }
+
+        /**
+         * Provides the Duration from the given duration URI.
+         * @param uri
+         * @return - Duration enum type
+         */
+        public static Duration getDurationFromURI(URI uri) {
```

Comment:
This could also return an Optional<Duration> if you would rather indicate it is invalid by showing no value exists for the URI instead of throwing an exception.

Reply:
This method has been removed.